File: generated_tests/pascal/tests/utilities_tests.pas
uses SplashKit, TestFramework, ../Helpers;
type
    TTestUtilities = class(TTestCase)
    protected
        procedure Setup; override;
    end;
begin
    inherited;
    SetResourcesPath('/mnt/c/Users/Noahc/Documents/aYear_2_semester_2/TeamProject/GitHubRepo/splashkit_test_generator/resources');
end;
procedure TestContainsIntegration;
begin
    AssertTrue(Contains('splashkit library', 'splashkit'));
    AssertTrue(Contains('splashkit library', 'library'));
    AssertTrue(Contains('splashkit library', 'it lib'));
    AssertTrue(Contains('splashkit library', ''));
    AssertFalse(Contains('splashkit library', 'unreal'));
    AssertFalse(Contains('', 'splashkit'));
end;
procedure TestConvertToDoubleIntegration;
begin
    AssertEquals(123.0, ConvertToDouble('123'));
    AssertEquals(123.456, ConvertToDouble('123.456'));
    AssertEquals(-123.0, ConvertToDouble('-123'));
    AssertEquals(-123.456, ConvertToDouble('-123.456'));
    AssertEquals(0.456, ConvertToDouble('.456'));
    AssertEquals(123.0, ConvertToDouble('123.'));
    AssertEquals(123.456, ConvertToDouble('000123.456'));
    AssertEquals(123.456, ConvertToDouble('123.456000'));
    AssertEquals(MaxDouble, ConvertToDouble('inf'));
    AssertEquals(-MaxDouble, ConvertToDouble('-inf'));
    AssertEquals(123.0, ConvertToDouble('1.23e2'));
end;
procedure TestConvertToIntegerIntegration;
begin
    AssertEquals(123, ConvertToInteger('123'));
    AssertEquals(123, ConvertToInteger('123.456'));
    AssertEquals(-123, ConvertToInteger('-123'));
    AssertEquals(-123, ConvertToInteger('-123.456'));
    AssertEquals(123, ConvertToInteger('123.'));
    AssertEquals(123, ConvertToInteger('000123.456'));
    AssertEquals(123, ConvertToInteger('123.456000'));
end;
procedure TestIndexOfIntegration;
begin
    AssertEquals(0, IndexOf('splashkit library', 'splashkit'));
    AssertEquals(10, IndexOf('splashkit library', 'library'));
    AssertEquals(7, IndexOf('splashkit library', 'it lib'));
    AssertEquals(-1, IndexOf('splashkit library', 'unreal'));
    AssertEquals(0, IndexOf('splashkit library', 'splashkit library'));
    AssertEquals(-1, IndexOf('splashkit library', 'splashkit library is the best'));
    AssertEquals(0, IndexOf('splashkit library', ''));
    AssertEquals(-1, IndexOf('', 'splashkit'));
    AssertEquals(0, IndexOf('', ''));
end;
procedure TestIsDoubleIntegration;
begin
    AssertTrue(IsDouble('123.456'));
    AssertTrue(IsDouble('123'));
    AssertTrue(IsDouble('-123.456'));
    AssertTrue(IsDouble('-123'));
    AssertTrue(IsDouble('123.'));
    AssertTrue(IsDouble('000123.456'));
    AssertTrue(IsDouble('123.456000'));
    AssertFalse(IsDouble('.456'));
    AssertFalse(IsDouble('SplashKit'));
    AssertFalse(IsDouble(''));
end;
procedure TestIsIntegerIntegration;
begin
    AssertTrue(IsInteger('123'));
    AssertTrue(IsInteger('-123'));
    AssertFalse(IsInteger('123.456'));
    AssertFalse(IsInteger('-123.456'));
    AssertFalse(IsInteger('.456'));
    AssertFalse(IsInteger('SplashKit'));
    AssertFalse(IsInteger(''));
end;
procedure TestIsNumberIntegration;
begin
    AssertTrue(IsNumber('123'));
    AssertTrue(IsNumber('123.456'));
    AssertTrue(IsNumber('-123'));
    AssertTrue(IsNumber('-123.456'));
    AssertFalse(IsNumber('SplashKit'));
    AssertFalse(IsNumber(''));
end;
procedure TestLengthOfIntegration;
begin
    AssertEquals(9, LengthOf('splashkit'));
    AssertEquals(0, LengthOf(''));
end;
procedure TestReplaceAllIntegration;
begin
    AssertEquals('SK library', ReplaceAll('splashkit library', 'splashkit', 'SK'));
    AssertEquals('splashkit lib', ReplaceAll('splashkit library', 'library', 'lib'));
    AssertEquals('splashkitlibrary', ReplaceAll('splashkit library', 'it lib', 'itlib'));
    AssertEquals('splashkit library', ReplaceAll('splashkit library', 'unreal', 'tournament'));
    AssertEquals('SK', ReplaceAll('splashkit library', 'splashkit library', 'SK'));
    AssertEquals(' library', ReplaceAll('splashkit library', 'splashkit', ''));
    AssertEquals('splashkit library', ReplaceAll('splashkit library', '', 'SK'));
    AssertEquals('', ReplaceAll('', '', 'SK'));
end;
procedure TestSplitIntegration;
begin
    result := Split('splashkit library', ' ');
    AssertEquals(TArray<String>.Create('splashkit', 'library'), result);
    AssertEquals(TArray<String>.Create(''), Split('', ' '));
    AssertEquals(TArray<String>.Create('splashkit library'), Split('splashkit library', ','));
    AssertEquals(TArray<String>.Create('', 'splashkit library'), Split(',splashkit library', ','));
    AssertEquals(TArray<String>.Create('splashkit library', ''), Split('splashkit library,', ','));
    AssertEquals(TArray<String>.Create('', 'splashkit library', ''), Split(',splashkit library,', ','));
    AssertEquals(TArray<String>.Create('splashkit', '', 'library'), Split('splashkit,,library', ','));
end;
procedure TestToLowercaseIntegration;
begin
    AssertEquals('splashkit', ToLowercase('SPLASHKIT'));
    AssertEquals('', ToLowercase(''));
    AssertEquals('splashkit', ToLowercase('splashkit'));
    AssertEquals('splashkit', ToLowercase('SpLaShKiT'));
end;
procedure TestToUppercaseIntegration;
begin
    AssertEquals('SPLASHKIT', ToUppercase('splashkit'));
    AssertEquals('', ToUppercase(''));
    AssertEquals('SPLASHKIT', ToUppercase('SPLASHKIT'));
    AssertEquals('SPLASHKIT', ToUppercase('SpLaShKiT'));
end;
procedure TestTrimIntegration;
begin
    AssertEquals('splashkit', Trim('  splashkit'));
    AssertEquals('splashkit', Trim('splashkit  '));
    AssertEquals('splashkit', Trim('  splashkit  '));
    AssertEquals('splashkit', Trim('splashkit'));
    AssertEquals('', Trim(''));
end;
procedure TestRndRangeIntegration;
begin
    AssertTrue((Rnd(-1, 5) >= -1) and (Rnd(-1, 5) <= 5));
    AssertEquals(1, Rnd(1, 1));
    AssertTrue((Rnd(5, 1) >= 1) and (Rnd(5, 1) <= 5));
end;
procedure TestRndIntegration;
begin
    AssertTrue((Rnd() >= 0.0) and (Rnd() <= MaxSingle));
end;
procedure TestRndIntIntegration;
begin
    AssertTrue((Rnd(1) >= 0) and (Rnd(1) <= 1));
    AssertTrue((Rnd(10) >= 0) and (Rnd(10) <= 10));
    AssertEquals(0, Rnd(-1));
    AssertEquals(0, Rnd(0));
end;
procedure TestCurrentTicksIntegration;
begin
    testTicks := CurrentTicks();
    AssertTrue(testTicks > 0);
end;
procedure TestDelayIntegration;
begin
    testTimer := CreateTimer('test_timer');
    CleanupTimer := TTimerCleanup.Create;
    StartTimer(testTimer);
    initialTicks := TimerTicks(testTimer);
    Delay(200);
    AssertTrue(initialTicks + 200 < TimerTicks(testTimer));
end;
procedure TestDisplayDialogIntegration;
begin
    testWindow := OpenWindow('Test Window', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    testFont := LoadFont('test_font', 'hara.ttf');
    CleanupFont := TFontCleanup.Create;
    DisplayDialog('Test Dialog', 'This is a test message', testFont, 20);
    FreeFont(testFont);
end;
procedure TestFileAsStringIntegration;
begin
    AssertEquals('BITMAP,ufo,ufo.png\n', FileAsString('blah.txt', ResourceKind.BUNDLE_RESOURCE));
    AssertEquals('', FileAsString('', ResourceKind.BUNDLE_RESOURCE));
    AssertEquals('', FileAsString('invalid.txt', ResourceKind.BUNDLE_RESOURCE));
end;
end;

procedure RegisterTests;
begin
    #<Proc:0x00007fbbcab52da8 /mnt/c/Users/Noahc/Documents/aYear_2_semester_2/TeamProject/GitHubRepo/splashkit_test_generator/test_generator/config/languages/pascal_config.rb:138 (lambda)>
end;

File: generated_tests/csharp/tests/utilities_tests.cs
using Xunit;
using static SplashKitSDK.SplashKit;

namespace SplashKitTests
{
    public class TestUtilities
    {
        [Fact]
        public void TestContainsIntegration()
        {
            Assert.True(Contains("splashkit library", "splashkit"));
            Assert.True(Contains("splashkit library", "library"));
            Assert.True(Contains("splashkit library", "it lib"));
            Assert.True(Contains("splashkit library", ""));
            Assert.False(Contains("splashkit library", "unreal"));
            Assert.False(Contains("", "splashkit"));
        }
        [Fact]
        public void TestConvertToDoubleIntegration()
        {
            Assert.Equal(123.0, ConvertToDouble("123"));
            Assert.Equal(123.456, ConvertToDouble("123.456"));
            Assert.Equal(-123.0, ConvertToDouble("-123"));
            Assert.Equal(-123.456, ConvertToDouble("-123.456"));
            Assert.Equal(0.456, ConvertToDouble(".456"));
            Assert.Equal(123.0, ConvertToDouble("123."));
            Assert.Equal(123.456, ConvertToDouble("000123.456"));
            Assert.Equal(123.456, ConvertToDouble("123.456000"));
            Assert.Equal(float.PositiveInfinity, ConvertToDouble("inf"));
            Assert.Equal(float.NegativeInfinity, ConvertToDouble("-inf"));
            Assert.Equal(123.0, ConvertToDouble("1.23e2"));
        }
        [Fact]
        public void TestConvertToIntegerIntegration()
        {
            Assert.Equal(123, ConvertToInteger("123"));
            Assert.Equal(123, ConvertToInteger("123.456"));
            Assert.Equal(-123, ConvertToInteger("-123"));
            Assert.Equal(-123, ConvertToInteger("-123.456"));
            Assert.Equal(123, ConvertToInteger("123."));
            Assert.Equal(123, ConvertToInteger("000123.456"));
            Assert.Equal(123, ConvertToInteger("123.456000"));
        }
        [Fact]
        public void TestIndexOfIntegration()
        {
            Assert.Equal(0, IndexOf("splashkit library", "splashkit"));
            Assert.Equal(10, IndexOf("splashkit library", "library"));
            Assert.Equal(7, IndexOf("splashkit library", "it lib"));
            Assert.Equal(-1, IndexOf("splashkit library", "unreal"));
            Assert.Equal(0, IndexOf("splashkit library", "splashkit library"));
            Assert.Equal(-1, IndexOf("splashkit library", "splashkit library is the best"));
            Assert.Equal(0, IndexOf("splashkit library", ""));
            Assert.Equal(-1, IndexOf("", "splashkit"));
            Assert.Equal(0, IndexOf("", ""));
        }
        [Fact]
        public void TestIsDoubleIntegration()
        {
            Assert.True(IsDouble("123.456"));
            Assert.True(IsDouble("123"));
            Assert.True(IsDouble("-123.456"));
            Assert.True(IsDouble("-123"));
            Assert.True(IsDouble("123."));
            Assert.True(IsDouble("000123.456"));
            Assert.True(IsDouble("123.456000"));
            Assert.False(IsDouble(".456"));
            Assert.False(IsDouble("SplashKit"));
            Assert.False(IsDouble(""));
        }
        [Fact]
        public void TestIsIntegerIntegration()
        {
            Assert.True(IsInteger("123"));
            Assert.True(IsInteger("-123"));
            Assert.False(IsInteger("123.456"));
            Assert.False(IsInteger("-123.456"));
            Assert.False(IsInteger(".456"));
            Assert.False(IsInteger("SplashKit"));
            Assert.False(IsInteger(""));
        }
        [Fact]
        public void TestIsNumberIntegration()
        {
            Assert.True(IsNumber("123"));
            Assert.True(IsNumber("123.456"));
            Assert.True(IsNumber("-123"));
            Assert.True(IsNumber("-123.456"));
            Assert.False(IsNumber("SplashKit"));
            Assert.False(IsNumber(""));
        }
        [Fact]
        public void TestLengthOfIntegration()
        {
            Assert.Equal(9, LengthOf("splashkit"));
            Assert.Equal(0, LengthOf(""));
        }
        [Fact]
        public void TestReplaceAllIntegration()
        {
            Assert.Equal("SK library", ReplaceAll("splashkit library", "splashkit", "SK"));
            Assert.Equal("splashkit lib", ReplaceAll("splashkit library", "library", "lib"));
            Assert.Equal("splashkitlibrary", ReplaceAll("splashkit library", "it lib", "itlib"));
            Assert.Equal("splashkit library", ReplaceAll("splashkit library", "unreal", "tournament"));
            Assert.Equal("SK", ReplaceAll("splashkit library", "splashkit library", "SK"));
            Assert.Equal(" library", ReplaceAll("splashkit library", "splashkit", ""));
            Assert.Equal("splashkit library", ReplaceAll("splashkit library", "", "SK"));
            Assert.Equal("", ReplaceAll("", "", "SK"));
        }
        [Fact]
        public void TestSplitIntegration()
        {
            var result = Split("splashkit library", ' ');
            Assert.Equal(new List<string> { "splashkit", "library" }, result);
            Assert.Equal(new List<string> { "" }, Split("", ' '));
            Assert.Equal(new List<string> { "splashkit library" }, Split("splashkit library", ','));
            Assert.Equal(new List<string> { "", "splashkit library" }, Split(",splashkit library", ','));
            Assert.Equal(new List<string> { "splashkit library", "" }, Split("splashkit library,", ','));
            Assert.Equal(new List<string> { "", "splashkit library", "" }, Split(",splashkit library,", ','));
            Assert.Equal(new List<string> { "splashkit", "", "library" }, Split("splashkit,,library", ','));
        }
        [Fact]
        public void TestToLowercaseIntegration()
        {
            Assert.Equal("splashkit", ToLowercase("SPLASHKIT"));
            Assert.Equal("", ToLowercase(""));
            Assert.Equal("splashkit", ToLowercase("splashkit"));
            Assert.Equal("splashkit", ToLowercase("SpLaShKiT"));
        }
        [Fact]
        public void TestToUppercaseIntegration()
        {
            Assert.Equal("SPLASHKIT", ToUppercase("splashkit"));
            Assert.Equal("", ToUppercase(""));
            Assert.Equal("SPLASHKIT", ToUppercase("SPLASHKIT"));
            Assert.Equal("SPLASHKIT", ToUppercase("SpLaShKiT"));
        }
        [Fact]
        public void TestTrimIntegration()
        {
            Assert.Equal("splashkit", Trim("  splashkit"));
            Assert.Equal("splashkit", Trim("splashkit  "));
            Assert.Equal("splashkit", Trim("  splashkit  "));
            Assert.Equal("splashkit", Trim("splashkit"));
            Assert.Equal("", Trim(""));
        }
        [Fact]
        public void TestRndRangeIntegration()
        {
            Assert.InRange(Rnd(-1, 5), -1, 5);
            Assert.Equal(1, Rnd(1, 1));
            Assert.InRange(Rnd(5, 1), 1, 5);
        }
        [Fact]
        public void TestRndIntegration()
        {
            Assert.InRange(Rnd(), 0.0, float.MaxValue);
        }
        [Fact]
        public void TestRndIntIntegration()
        {
            Assert.InRange(Rnd(1), 0, 1);
            Assert.InRange(Rnd(10), 0, 10);
            Assert.Equal(0, Rnd(-1));
            Assert.Equal(0, Rnd(0));
        }
        [Fact]
        public void TestCurrentTicksIntegration()
        {
            var testTicks = CurrentTicks();
            Assert.True(testTicks > 0);
        }
        [Fact]
        public void TestDelayIntegration()
        {
            var testTimer = CreateTimer("test_timer");
            StartTimer(testTimer);
            var initialTicks = TimerTicks(testTimer);
            Delay(200);
            Assert.True(initialTicks + 200 < TimerTicks(testTimer));
        }
        [Fact]
        public void TestDisplayDialogIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testFont = LoadFont("test_font", "hara.ttf");
            DisplayDialog("Test Dialog", "This is a test message", testFont, 20);
            FreeFont(testFont);
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestFileAsStringIntegration()
        {
            Assert.Equal("BITMAP,ufo,ufo.png\n", FileAsString("blah.txt", ResourceKind.BundleResource));
            Assert.Equal("", FileAsString("", ResourceKind.BundleResource));
            Assert.Equal("", FileAsString("invalid.txt", ResourceKind.BundleResource));
        }
    }
}
